codeops-scripts/apply-github-settings.ps1

```diff
@@ -175,11 +190,30 @@ function _main {
         }
     }
 
-    if ($script:failedRepos.Count -gt 0) {
-        $ErrorActionPreference = "Continue"
-        $errorMessage = "The following repositories reported errors during processing:`n{0}" -f ($script:failedRepos -join "`n")
-        Write-Error $errorMessage
-        exit 1
+    $runMetadata += @{ end_time = [datetime]::UtcNow }
+    $runResults += @{ metadata = $runMetadata }
+    $runResults += @{ orgs = $allOrgResults }
+
+    # Produce a JSON report file
+    $reportFile = "apply-github-settings-report.json"
+    $runResults | ConvertTo-Json -Depth 30 | Out-File $reportFile -Force
+
+    # Upload JSON report to datalake
+    if ($env:DATALAKE_NAME -and $env:DATALAKE_SASTOKEN -and $env:DATALAKE_FILESYSTEM -and $env:DATALAKE_DIRECTORY) {
+        Write-Host "Publishing report to datalake: $($env:DATALAKE_NAME)"
+        $timestamp = (Get-Date).ToUniversalTime().ToString('yyyyMMddHHmmssfff')
```

Comment:
I notice you have this, and also up on line 193 you're using `[datetime]::UtcNow`. I'm curious to know whether there's a reason for using these two different mechanism?

But the main thing I wanted to ask is whether it might matter that you've got two different "nows" in play here. The `$runMetadata` will contain whatever the timestamp captured up there was, and then this one will be a tad later. Often things run fast enough that when you fetch "now" multiple times in quick succession you get the same result, but sometimes you won't, and I'm wondering if there's any possibility that this sort of "slicing" of now could cause a problem?

(The classic one is where people read `UtcNow` multiple times when formatting a message, which, if you do it around midnight, can result in really confusing results, because you might pick up the day just before midnight but the hour just after, causing a result that's wrong by around 24 hours. You're not doing that particular mistake here, but in general, if I need several strings or other values based on some notion of "now" I usually try to read the current time into a variable exactly once, and then work from that.)

Reply:
I hadn't expected the filename to be used for anything in particular, but that's probably just a failure of imagination on my part, I've updated it to use the `metadata.start_time`.